core/instance.go

```diff
@@ -269,6 +269,9 @@ func (i *instance) getCheapestCompatibleSpotInstanceType(allowedList []string, d
 
 		candidatePrice := i.calculatePrice(candidate)
 
+		if len(candidate.virtualizationTypes) == 0 {
+			candidate.virtualizationTypes = append(candidate.virtualizationTypes, "HVM")
+		}
```

Comment:
I'd rather have that move to [the method comparing virtualization](https://github.com/cristim/autospotting/blob/master/core/instance.go#L213) - because it's directly related to visualization comparison, and if we have more exception, this method will become unmaintainable.

Reply:
Also there are no tests for this condition.

Reply:
Yeah that makes sense. I moved it and added a test case for it.